src/anniversary/sf33rd/Source/Game/RAMCNT.c

```diff
@@ -0,0 +1,185 @@
+#include "sf33rd/Source/Game/RAMCNT.h"
+#include "common.h"
+#include "sf33rd/AcrSDK/ps2/flps2debug.h"
+#include "sf33rd/AcrSDK/ps2/foundaps2.h"
+#include "sf33rd/Source/Common/MemMan.h"
+#include "sf33rd/Source/Game/debug/Debug.h"
+#include "sf33rd/Source/Game/texgroup.h"
+
+#define ERR_STOP                                                                                                       \
+    do {                                                                                                               \
+    } while (1)
+
+void disp_ramcnt_free_area() {
+    if (Debug_w[0xA]) {
+        flPrintColor(-0x71U);
+        flPrintL(4, 8, "Ramcnt Status");
+        flPrintL(4, 9, "Now %07X", mmGetRemainder(&rckey_mmobj));
+        flPrintL(4, 0xA, "Min %07X", mmGetRemainderMin(&rckey_mmobj));
+        flPrintL(4, 0xB, "Key %2d / %2d", rckeymin, rckeyctr);
+    }
+}
+
+void Init_ram_control_work(u8 *adrs, s32 size) {
+    s16 i;
+
+    mmHeapInitialize(&rckey_mmobj, adrs, size, 0x40, "- for Ramcnt -");
+
+    for (i = 0; i < 0x3f; i++) {
+        rckeyque[i] = (0x3F - i);
+    }
+
+    rckeyctr = 0x3F;
+    rckeymin = 0x3F;
+    rckeyque[rckeyctr] = 0;
+
+    for (i = 0; i < sizeof(RCKeyWork) / 4; i++) {
+        ((s32 *)rckey_work)[i] = 0;
+    }
+
+    for (i = 1; i < 0x40; i++) {
+        rckey_work[i] = rckey_work[0];
+    }
+}
+
+void Push_ramcnt_key(s16 key) {
+    RCKeyWork *rwk;
+
+    rwk = &rckey_work[key];
+    if (rwk->use != 0) {
+        if ((rwk->type == 8) || (rwk->type == 9)) {
+            flLogOut("TEXCASH KEY PUSH ERROR\n");
+            ERR_STOP;
+        }
+        Push_ramcnt_key_original_2(key);
+    }
+    return;
+}
+
+void Push_ramcnt_key_original(s16 key) {
+    RCKeyWork *rwk;
+
+    rwk = &rckey_work[key];
+    if (rwk->use != 0) {
+        if ((rwk->type != 8) && (rwk->type != 9)) {
+            flLogOut("TEXCASH KEY PUSH ERROR2\n");
+            ERR_STOP;
+        }
+        Push_ramcnt_key_original_2(key);
+    }
+    return;
+}
+
+void Push_ramcnt_key_original_2(s16 key) {
+    RCKeyWork *rwk;
+
+    rwk = &rckey_work[key];
+    if (rwk->use != 0) {
+        mmFree(&rckey_mmobj, (u8 *)rwk->adr);
+        rwk->type = 0;
+        rwk->use = 0;
+        if (rwk->group_num) {
+            purge_texture_group(rwk->group_num);
+        }
+        rwk->group_num = 0;
+        rckeyque[rckeyctr++] = key;
+    }
+}
+
+void Purge_memory_of_kind_of_key(u8 kokey) {
+    RCKeyWork *rwk;
+    s16 i;
+
+    for (i = 0; i < 0x40; i += 1) {
+        rwk = &rckey_work[i];
+        if (rwk->use && (rwk->type == kokey)) {
+            Push_ramcnt_key(i);
+        }
+    }
+}
+
+void Set_size_data_ramcnt_key(s16 key, u32 size) {
+    if (key <= 0) {
+        flLogOut("未使用のメモリキーへファイルサイズを格納しようとしました。\n");
```

Comment:
Can you also please add translations for japanese strings? That'll make reading this code in the future much more straightforward
```suggestion
        // An attempt was made to store a file size in an unused memory key.\n
        flLogOut("未使用のメモリキーへファイルサイズを格納しようとしました。\n");
```

Reply:
Fair point. Noted for any future strings like that.

Reply:
Yes, that's what I usually do